tools/docker-build/Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM ubuntu:24.04 as build
+
+RUN apt-get update && apt-get install -y build-essential software-properties-common git git-lfs gettext cmake mtools wget curl which
+
+ARG ARM_TOOLCHAIN_URL=https://developer.arm.com/-/media/Files/downloads/gnu/13.2.rel1/binrel/
+ARG ARM_TOOLCHAIN_FILE=arm-gnu-toolchain-13.2.rel1-x86_64-arm-none-eabi.tar.gz
+ARG DOWNLOAD_DIR=/root/bin
+
+RUN mkdir ${DOWNLOAD_DIR} && cd ${DOWNLOAD_DIR} && curl -LO ${ARM_TOOLCHAIN_URL}/${ARM_TOOLCHAIN_FILE}
+
+RUN cd ${DOWNLOAD_DIR} && pwd && tar -xf ${ARM_TOOLCHAIN_FILE}
+
+ARG ARM_TOOLCHAIN_DIR=arm-gnu-toolchain-13.2.Rel1-x86_64-arm-none-eabi
+
+ENV PATH="${PATH}:${DOWNLOAD_DIR}/${ARM_TOOLCHAIN_DIR}/bin"
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -y | sh
+
+RUN apt-get install -y git wget flex bison gperf python3 python3-pip python3-venv cmake ninja-build
+
+RUN apt-get install -y ccache libffi-dev libssl-dev dfu-util libusb-1.0-0 python-is-python3
+
+ENV IDF_TOOLS_PATH=/root/.espressif
+ENV IDF_PATH=/root/esp/esp-idf
+
+RUN mkdir -p /root/esp && cd /root/esp && git clone -b v5.2.2 --recursive https://github.com/espressif/esp-idf.git && ls
+
+RUN cd /root/esp/esp-idf && ./install.sh all
+
+RUN curl -LO https://raw.githubusercontent.com/adafruit/circuitpython/main/requirements-dev.txt
+
+RUN curl -LO https://raw.githubusercontent.com/adafruit/circuitpython/main/requirements-doc.txt
```

Comment:
This gets the requirements files from the tip of main, but I think you want to get them from the repo?

Reply:
The `Dockerfile` is only used to build the image.   I'm including it so that folks can see what it is doing (basically the same steps you would need to do on the command line to install the tools) and be empowered to create their own images.

Reply:
> This gets the requirements files from the tip of main, but I think you want to get them from the repo?

You mean the repo that CP is being built from?  Can't do that because the image is given access to that repo at runtime, but the requirements need to be available at image build time so that the python built into the image can be configured.  If you wanted to do it the other way, you'd need to have python pull in the requirements every time it is asked to build CP.